# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1429398422 18000
#      Sat Apr 18 18:07:02 2015 -0500
# Node ID ce027487af26e83160196a15f4c40c7ee215c426
# Parent  13315376edaa31f0f38d6983f39728220ca36cd0
Remove extraneous border from background rectangles for gnuplot tk (bug #44871).

* __go_draw_figure__.m (__go_draw_figure__): Add option "fs solid noborder" to
the obj 1 (figure) and obj 2 (axis) rectangle specifications.

diff --git a/scripts/plot/util/private/__go_draw_figure__.m b/scripts/plot/util/private/__go_draw_figure__.m
--- a/scripts/plot/util/private/__go_draw_figure__.m
+++ b/scripts/plot/util/private/__go_draw_figure__.m
@@ -33,17 +33,17 @@ function __go_draw_figure__ (h, plot_str
 
     if (nkids > 0)
       fputs (plot_stream, "\nreset;\n");
       fputs (plot_stream, "set autoscale keepfix;\n");
       fputs (plot_stream, "set origin 0, 0\n");
       fputs (plot_stream, "set size 1, 1\n");
       bg = get (h, "color");
       if (isnumeric (bg))
-        fprintf (plot_stream, "set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\"\n", round (255 * bg));
+        fprintf (plot_stream, "set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\" fs solid noborder\n", round (255 * bg));
         bg_is_set = true;
       else
         bg_is_set = false;
       endif
       fg_was_set = false;
 
       for i = nkids:-1:1
         type = get (kids(i), "type");
@@ -135,17 +135,17 @@ function __go_draw_figure__ (h, plot_str
               ## Rely upon listener to convert axes position
               ## to "normalized" units.
               orig_axes_units = get (kids(i), "units");
               orig_axes_position = get (kids(i), "position");
               unwind_protect
                 set (kids(i), "units", "normalized");
                 fg = get (kids(i), "color");
                 if (isnumeric (fg) && strcmp (get (kids(i), "visible"), "on"))
-                  fprintf (plot_stream, "set obj 2 rectangle from graph 0,0 to graph 1,1 behind fc rgb \"#%02x%02x%02x\"\n", round (255 * fg));
+                  fprintf (plot_stream, "set obj 2 rectangle from graph 0,0 to graph 1,1 behind fc rgb \"#%02x%02x%02x\" fs solid noborder\n", round (255 * fg));
                   fg_is_set = true;
                   fg_was_set = true;
                 elseif (fg_was_set)
                   fprintf (plot_stream, "unset obj 2\n");
                   fg_is_set = false;
                   fg_was_set = false;
                 else
                   fg_is_set = false;
